CONTRIBUTING.md

```diff
@@ -150,29 +156,45 @@ If you are so inclined, you can open a draft PR as you continue to work on it. S
 ## Tests 🧪
 ### Writing Browser/System/Feature Tests/Specs
 
-Add a test for your change. If you are adding functionality or fixing a  bug, you should add a test!
-
-If you are inexperienced in writing tests or get stuck on one, please reach out for help :). You probably don't need to write new tests when simple re-stylings are done (ie. the page may look slightly different but the Test suite is unaffected by those changes).
-
-If you need to see a browser/system spec run in the browser, you can use the following env variable:
+Add a test for your change. If you are adding functionality or fixing a bug, you should add a test!
+
+If you are inexperienced in writing tests or get stuck on one, please reach out for help :)
+
+#### Guidelines
+- When creating factories, in each RSpec test, hard code all values that you check with a RSpec matcher. Don't check FactoryBot default values. See [#4217](https://github.com/rubyforgood/human-essentials/issues/4217) for why.
+- Write tests to pass with Event Sourcing turned both on and off, see the [Event Sourcing wiki page](https://github.com/rubyforgood/human-essentials/wiki/Event-Sourcing).
+- Keep individual tests tightly scoped, only test the endpoint that you want to test. E.g. create inventory directly using `TestInventory` rather than using an additional endpoint.
```

Comment:
Can we add a line about not adding system tests unless we actually need to test interactivity (e.g. JavaScript) and preferring request tests instead?

Reply:
resolved in [b5e13e5](https://github.com/rubyforgood/human-essentials/pull/4551/commits/b5e13e570aae8b4d392ee5032c289ebdb559566c)

also added wiki contribution workflow and fixed dev container instructions